site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java

```diff
@@ -13,48 +11,59 @@
 import org.hippoecm.hst.core.parameters.ParametersInfo;
 import org.hippoecm.hst.core.request.HstRequestContext;
 import org.hippoecm.hst.util.SearchInputParsingUtils;
+import org.onehippo.cms7.essentials.components.EssentialsSearchComponent;
+import org.onehippo.cms7.essentials.components.info.EssentialsListComponentInfo;
+import org.onehippo.cms7.essentials.components.paging.Pageable;
 import uk.nhs.digital.ps.beans.Publication;
 
 import static org.hippoecm.hst.content.beans.query.builder.ConstraintBuilder.constraint;
 import static org.hippoecm.hst.content.beans.query.builder.ConstraintBuilder.or;
 
-@ParametersInfo(type = SearchComponentInfo.class)
-public class SearchComponent extends BaseHstComponent {
+@ParametersInfo(type = EssentialsListComponentInfo.class)
+public class SearchComponent extends EssentialsSearchComponent {
 
     @Override
     public void doBeforeRender(final HstRequest request, final HstResponse response) throws HstComponentException {
 
         HstRequestContext requestContext = request.getRequestContext();
         HippoBean scopeBean = requestContext.getContentBean();
-        SearchComponentInfo info = getComponentParametersInfo(request);
+        final EssentialsListComponentInfo paramInfo = getComponentParametersInfo(request);
         String query = SearchInputParsingUtils.parse(
             getPublicRequestParameter(request, "query"), false);
-        int offset = NumberUtils.toInt(
-            getPublicRequestParameter(request, "offset"), 0);
+        final int pageSize = getPageSize(request, paramInfo);
+        final int currentPage = getCurrentPage(request);
```

Comment:
`currentPage` suggests that this variable represents some object of type `Page`. How about renaming it to something like `currentPageNumber`, `currentPageIndex` or `currentPageOrdinal`?

Reply:
Good idea. will go for currentPageNumber